qt/widgets/common/src/WorkspacePresenter/WorkspaceTreeWidgetSimple.cpp

```diff
@@ -185,7 +192,41 @@ void WorkspaceTreeWidgetSimple::popupContextMenu() {
     } else if (std::dynamic_pointer_cast<IMDWorkspace>(workspace)) {
       menu->addAction(m_showAlgorithmHistory);
       menu->addAction(m_sampleLogs);
-      menu->addAction(m_sliceViewer);
+
+      // launch slice viewer or plot spectrum conditionally
+      bool add_slice_viewer = true;
+
+      if (std::dynamic_pointer_cast<IMDHistoWorkspace>(workspace)) {
+        auto mdhist_ws =
+            std::dynamic_pointer_cast<IMDHistoWorkspace>(workspace);
+        // if the number of non-integral  if the number of non-integrated
+        // dimensions is 1.
+        auto num_dims = mdhist_ws->getNumDims();
+        if (num_dims == 1) {
+          // only 1 dimension
+          add_slice_viewer = false;
+        } else {
+          // multiple dimension: check number of non-integral
+          auto num_no_integral_dim = 0;
```

Comment:
This should probably be replaced with `mdhist_ws->getNonIntegratedDimensions().size()`